plugins/com.google.cloud.tools.eclipse.ui.util/src/com/google/cloud/tools/eclipse/ui/util/event/OpenUriSelectionListener.java

```diff
@@ -42,35 +45,37 @@ public void widgetDefaultSelected(SelectionEvent event) {
 
   private void openAppEngineDashboard(String uriString) {
     try {
-      String projectId = projectIdProvider.getProjectId();
-      URI uri = appendProjectId(new URI(uriString), projectId);
+      URI uri = appendQueryParameters(new URI(uriString));
       browserSupport.getExternalBrowser().openURL(uri.toURL());
     } catch (PartInitException | MalformedURLException | URISyntaxException ex) {
       errorHandler.handle(ex);
     }
   }
 
-  private URI appendProjectId(URI uri, String projectId) throws URISyntaxException {
-    if (projectId != null && !projectId.isEmpty() && !projectId.trim().isEmpty()) {
-      projectId = projectId.trim();
-      String query = uri.getQuery();
-      if (query == null) {
-        query = URI_PARAM_PROJECT + "=" + projectId;
-      } else {
-        query += "&" + URI_PARAM_PROJECT + "=" + projectId;
+  private URI appendQueryParameters(URI uri) throws URISyntaxException {
+    String queryString = uri.getQuery();
+    if (queryString == null) {
+      queryString = "";
+    }
+    StringBuilder query = new StringBuilder(queryString);
+    for (Entry<String, String> parameter : queryParameterProvider.getParameters().entrySet()) {
+      if (query.length() > 0) {
+        query.append('&');
       }
-      return new URI(uri.getScheme(), uri.getUserInfo(), uri.getHost(),
-                     uri.getPort(), uri.getPath(), query, uri.getFragment());
-    } else {
-      return uri;
+      query.append(parameter.getKey())
```

Comment:
Don't we need to ensure the parameter name and values are URL encoded?  (e.g., in case the key contains a `=`, or the value contains a `&`)